src/lib/sap_ha/system/local.rb

```diff
@@ -144,10 +147,12 @@ def write_csync2_key(data)
             "but the key data is empty"
           return false
         end
-        status = pipe(['echo', data], ['uudecode', '-o', CSYNC2_KEY_PATH])
-        NodeLogger.log_status(status,
-          "Wrote the shared csync2 authentication key",
-          "Could not write the shared csync2 authentication key")
+        begin
+          File.write(CSYNC2_KEY_PATH, Base64.decode64(data))
+          NodeLogger.info("Wrote the shared csync2 authentication key")
+        rescue
+          NodeLogger.error("Could not write the shared csync2 authentication key")
```

Comment:
well, generic rescue is ok, but I would at least log what was issue. So something like
```suggestion
        rescue => error
          NodeLogger.error("Could not write the shared csync2 authentication key. Error: #{error}")
```